soccer/vision/camera/Camera.cpp

```diff
@@ -0,0 +1,389 @@
+    for (KalmanRobot& kalmanRobot : singleKalmanRobotList) {
+        std::list<CameraRobot>& measurementRobot = appliedRobotsList.at(kalmanRobotIdx);
+
+        int cameraRobotIdx = 0;
+        for (const CameraRobot& cameraRobot : singleRobotList) {
+            double dist = (kalmanRobot.getPos() - cameraRobot.getPos()).mag();
+
+            // Increase the distance of our cutoff by the velocity
+            // This is so the robot doesn't move outside the kalman filter position radius when the robot
+            // instantly stops (like in sim)
+            if (dist < *MHKF_radius_cutoff + kalmanRobot.getVel().mag()) {
+                measurementRobot.push_back(cameraRobot);
+                usedCameraRobot.at(cameraRobotIdx) = true;
+            }
+            cameraRobotIdx++;
+        }
+
+        kalmanRobotIdx++;
+    }
+
+    // Predict and update the filters based on measurements
+    kalmanRobotIdx = 0;
+    for (KalmanRobot& kalmanRobot : singleKalmanRobotList) {
+        std::list<CameraRobot>& measurementRobots = appliedRobotsList.at(kalmanRobotIdx);
+
+        // We had at least one measurement near this Robot
+        if (measurementRobots.size() > 0) {
+            CameraRobot avgRobot = CameraRobot::CombineRobots(measurementRobots);
+            kalmanRobot.predictAndUpdate(calcTime, avgRobot);
+
+        // There aren't any measurements so just predict
+        } else {
+            kalmanRobot.predict(calcTime);
+        }
+    }
+
+    // Create kalman robots if one isn't near camera measurement
+    int cameraRobotIdx = 0;
+    for (const CameraRobot& cameraRobot : singleRobotList) {
+        bool wasUsed = usedCameraRobot.at(cameraRobotIdx);
+
+        if (!wasUsed && singleKalmanRobotList.size() < *max_num_kalman_robots) {
+            singleKalmanRobotList.emplace_back(cameraID, calcTime, cameraRobot, previousWorldRobot);
+        }
+        
+        cameraRobotIdx++;
+    }
+}
+
+void Camera::updateRobotsAKF(RJ::Time calcTime,
+                             const std::list<CameraRobot>& singleRobotList,
+                             const WorldRobot& previousWorldRobot,
+                             std::list<KalmanRobot>& singleKalmanRobotList) {
+    
+    std::list<CameraRobot> singleRobotList2;
+    std::copy(singleRobotList.begin(), singleRobotList.end(), std::back_inserter(singleRobotList2));
+
+    // If we have no existing filters, create a new one from average of everything
+    if (singleKalmanRobotList.size() == 0) {
+        CameraRobot avgRobot = CameraRobot::CombineRobots(singleRobotList2);
+        singleKalmanRobotList.emplace_back(cameraID, calcTime, avgRobot, previousWorldRobot);
+
+        return;
+    }
+
+    // Average everything and add as measuremnet
+    CameraRobot avgRobot = CameraRobot::CombineRobots(singleRobotList2);
+    // Kinda cheating, but we are only keeping a single element in the list
+    singleKalmanRobotList.front().predictAndUpdate(calcTime, avgRobot);
```

Comment:
Should you remove the rest of the list if you're going to ignore the other estimates in the future? (Or set them to unhealthy and let the autoremoval code take care of it?)

Reply:
There should only be a single item in the list. We only add one if there's nothing in the list, then we only refer to the first one while it's still valid. 

In the case that it swaps from MHEKF to the AKF method, all of the other filters should die within a half a second, and this one should move to the correct state somewhere within a second or two. I think it's reasonable to assume that if someone is swapping the config values, it's ok if the estimates are bad for a short time.